src/NUnitFramework/framework/Internal/Commands/BeforeAndAfterTestCommand.cs

```diff
@@ -54,27 +54,21 @@ public override TestResult Execute(TestExecutionContext context)
 
             try
             {
-                BeforeTest(context);
-
-                context.CurrentResult = innerCommand.Execute(context);
-            }
-            catch (Exception ex)
-            {
-#if !NETSTANDARD1_6
-                if (ex is ThreadAbortException)
-                    Thread.ResetAbort();
-#endif
-                context.CurrentResult.RecordException(ex);
+                context.ExecuteWithExceptionHandling(c =>
```

Comment:
This make the `TestExecutionContext` know something about how Commands work and how to handle exceptions. I've been slowly moving in the opposite direction: Commands know about Context and not the other way around. I have to admit it's hard to make gradual changes when the project now has lots of participants and might be a bad idea. Something for my next project. 😄

Reply:
This is that helper method which I can't see any way around. It doesn't have to be on `TestExecutionContext` for sure, but it needs to be somewhere. See https://github.com/nunit/nunit/pull/2565#discussion_r155956395.

Reply:
Also: we definitely needed to deduplicate the code which ended up in this helper method. Leaving things the way they were is not an option.